platform/mbed_error.c

```diff
@@ -29,34 +29,18 @@
 #include <stdio.h>
 #endif
 
-//Helper macro to get the current SP
-#define GET_CURRENT_SP(sp)                                                          \
```

Comment:
Huh. Why is this macro no longer needed?

Reply:
SP pointer is fetched using the address of local variable, hence not needed

Reply:
It was a very complicated way of getting the current stack pointer.

Although, on reflection, I now think the current stack pointer is not what you wanted anyway. It was being used as part of a "show current thread" line, where the current SP is being shown next to the stack size and limit for the current thread.

In a fault handler, both the complex and simple version will give you MSP, which is not going to be anywhere near the thread stack. You really wanted the PSP (or SP_usr for Cortex-A).

I'll probably adjust it again in a further patch.